src/Tests/FlatSharpCompilerTests/UnionTests.cs

```diff
@@ -109,5 +109,20 @@ struct ValueStruct (fs_valueStruct) {{ x : int; }}
             new()));
 
         Assert.Contains("FlatSharp unions may not contain duplicate types. Union = UnionTests.MyUnion", ex.Message);
+    }
+
+    [Fact]
+    public void Union_WithCustomEnumValue()
+    {
+        string schema = @"
+            namespace UnionTests;
+            table A {}
```

Comment:
This syntax is new to me. This is supposed to be assigning custom discriminator values for each union member, right? Assuming that's the case, I don't understand why all the indexes are assumed to start at 1 in ReferenceUnionSchemaModel

Reply:
Yes, flatbuffer supports specifying custom enum values for each candidate option of the union. If not specified, the values will default to incrementing from 1 onwards. This is because in the implementation of flatbuffer, there is a default candidate option "None" for the union, with an enum value of 0.

Reply:
Got it, thanks! It makes more sense after checking out your branch and playing with it. Before merging, I'd like to add a couple more test cases:

```c#

    [TestMethod]
    public void UnionWithCustomEnumValues_Accept()
    {
        C c = new C(new A());

        Assert.AreEqual(2, c.Discriminator);
        Assert.AreEqual(C.ItemKind.A, c.Kind);

        Assert.IsNotNull(c.Item1);
        Assert.ThrowsException<InvalidOperationException>(() => c.Item2);

        Assert.AreEqual("A", c.Accept<Visitor, string>(new()));

        Assert.AreEqual(
            "A", 
            c.Match(
                (A a) => "A",
                (B b) => "B"));

        c = new C(new B());

        Assert.AreEqual(4, c.Discriminator);
        Assert.AreEqual(C.ItemKind.B, c.Kind);

        Assert.IsNotNull(c.Item2);
        Assert.ThrowsException<InvalidOperationException>(() => c.Item1);

        Assert.AreEqual("B", c.Accept<Visitor, string>(new()));

        Assert.AreEqual(
            "B",
            c.Match(
                (A a) => "A",
                (B b) => "B"));
    }

    [TestMethod]
    public void UnionWithCustomEnumValues_Clone()
    {
        Outer outer = new()
        {
            Union = new(new B())
        };

        Outer parsed = outer.SerializeAndParse();
        Assert.AreEqual(4, parsed.Union.Value.Discriminator);

        Outer cloned = new Outer(parsed);
        Assert.AreEqual(4, cloned.Union.Value.Discriminator);
    }
```

```idl
attribute "fs_serializer";

namespace FlatSharpEndToEndTests.UnionWithCustomEnumValues;

table A {}
table B {}
union C { A = 2, B = 4 }

table Outer (fs_serializer)
{
    union : C;
}
```

Thanks again!